reconcile/cli.py

```diff
@@ -369,26 +384,37 @@ def run_integration(func_container, ctx, *args, **kwargs):
         sys.stderr.write("Integration missing QONTRACT_INTEGRATION.\n")
         sys.exit(ExitCodes.ERROR)
 
-    try:
-        gql.init_from_config(
-            sha_url=ctx["gql_sha_url"],
-            integration=int_name,
-            validate_schemas=ctx["validate_schemas"],
-            print_url=ctx["gql_url_print"],
-        )
-    except GqlApiIntegrationNotFound as e:
-        sys.stderr.write(str(e) + "\n")
-        sys.exit(ExitCodes.INTEGRATION_NOT_FOUND)
-
     unleash_feature_state = get_feature_toggle_state(int_name)
     if not unleash_feature_state:
         logging.info("Integration toggle is disabled, skipping integration.")
         sys.exit(ExitCodes.SUCCESS)
 
     dry_run = ctx.get("dry_run", False)
+    early_exit_compare_sha = ctx.get("early_exit_compare_sha")
 
     try:
-        func_container.run(dry_run, *args, **kwargs)
+        if early_exit_compare_sha:
```

Comment:
there is no `else` here, so we will never run if this param is not set ?

Reply:
[49883f4](https://github.com/app-sre/qontract-reconcile/pull/2672/commits/49883f48add320f7ec73933c80efae056ab63c6f)